backend/logger/logger.py

```diff
@@ -0,0 +1,92 @@
+from fileHandler import FileHandler
+from singleton import Singleton
+
+from datetime import datetime
+import os
+from inspect import stack, getframeinfo
+
+
+class Logger(metaclass=Singleton):
+    """
+    Main class for logging functionalities.
+    """
+
+    def __init__(self, file_name="session", level=0):
```

Comment:
I think we should have constants for the log levels, such as CRITICAL=0, WARN=1, INFO=2, DEBUG=3 etc; I am not too familiar with the numbers representing log levels. Or at least in the docstring of this function

Reply:
Agreed.

Reply:
I think using an enum might be good for that